### DISCUSSION TITLE
Workaround for "Failed to start ContainerManager" err="failed to get rootfs info: failed to get device for dir ..." ?

### DISCUSSION THREAD

**Discussion by dev-zero** [k3s-io/k3s#7006] [Q&A]
Hi,

I am trying to run K3s with `--rootless` and my home directory is on a Lustre filesystem mount. Unless I use `-d ...` to move it to a node-local location (ex. `/tmp/...`) I get the error

```
"Failed to start ContainerManager" err="failed to get rootfs info: failed to get device for dir ..."
```

For *kind* I found the [instructions](https://github.com/kubernetes-sigs/kind/blob/main/site/content/docs/user/known-issues.md#failed-to-get-rootfs-info--stat-failed-on-dev) to manually specify the volume. How would I do that with K3s?
Or how can I debug this further? Maybe the required ops are not even possible with Lustre as a home dir?

**Comment by brandond** [comment]
Are you running k3s in docker? I'm confused why you would mention kind if not.

**Reply by dev-zero** [reply]
No, I am not. But this is the only mentioning of a workaround/fix I found while searching the internet.
All other reported cases of this error I saw are either unresolved or intermittent failures due to other problems.

**Reply by brandond** [reply]
I'm not sure anyone's ever tested containerd on Lustre. I suspect you'll need to play with setting the `--snapshotter` to something else that works on lustre, or put the data directory on a more traditional filesystem that supports overlayfs.

**Reply by dev-zero** [reply]
I will try that, thanks. Inspecting the source of the underlying functions to identify the device indicates that only an explicit list of filesystems are permitted.

The function doing the dir to device mapping:
https://github.com/google/cadvisor/blob/98c64146a370a6221ae1cc26e862ec2ff785af13/fs/fs.go#L572-L603

The partitions map population:
https://github.com/google/cadvisor/blob/98c64146a370a6221ae1cc26e862ec2ff785af13/fs/fs.go#L171-L237
